lib/puppet/reports/http.rb

```diff
@@ -14,7 +14,7 @@
   def process
     url = URI.parse(Puppet[:reporturl])
     headers = { "Content-Type" => "application/x-yaml" }
-    options = {}
+    options = { :metric_id => [:puppet, :report, :http] }
```

Comment:
@rlinehan based on my reading of https://github.com/puppetlabs/puppet/blob/master/lib/puppet/network/http/connection.rb#L25-L54 I think we may need to extend the `Puppet::Network::HTTP::Connection` class to be aware of the `:metric_id` option for this to not fail. I may certainly be missing something though..